### PR TITLE
great commit

### PR DESCRIPTION
**Pull request by Main444**
great commit

### CODE REVIEW

File: src/main/java/com/pharmacy/optican/demo/service/UserService.java [hunk 1]
```diff
@@ -3,96 +3,112 @@
...
+    public User findUserByPhone(String phone) {
         Optional<User> user = userRepository.findUserByPhone(phone);
         return user.orElse(null);
     }
 
     @PreAuthorize("isAuthenticated()")
-    public void updateUser(User user){
+    public void updateUser(User user) {
+
         Optional<User> oldUser = userRepository.findById(user.getId());
```
Comment (EvgenKalashnikov) [inline src/main/java/com/pharmacy/optican/demo/service/UserService.java:66]: add NPE check for user

File: src/main/java/com/pharmacy/optican/demo/service/UserService.java [hunk 2]
```diff
@@ -3,96 +3,112 @@
...
         oldUser.ifPresent(u -> {
 
-            if(!user.getEmail().isEmpty() && !user.getEmail().equals(u.getEmail())){
-                u.setEmail(user.getEmail());
+            if (!StringUtils.isBlank(user.getEmail()) && !user.getEmail().equals(u.getEmail()) && validateEmail(user.getEmail())) {
+                u.setEmail(user.getEmail().trim().toLowerCase());
+            }
+            if (!StringUtils.isBlank(user.getPhone()) && !user.getPhone().equals(u.getPhone()) && validatePhone(user.getPhone())) {
+                u.setPhone(user.getPhone().trim());
```
Comment (EvgenKalashnikov) [inline src/main/java/com/pharmacy/optican/demo/service/UserService.java]: romove all blank spaces

File: src/main/java/com/pharmacy/optican/demo/service/UserService.java [hunk 3]
```diff
@@ -3,96 +3,112 @@
...
             }
-            if(!user.getPhone().isEmpty() && !user.getPhone().equals(u.getPhone())){
-                u.setPhone(user.getPhone());
+            if (user.getFullName() != null && !user.getFullName().equals(u.getFullName())) {
```
Comment (EvgenKalashnikov) [inline src/main/java/com/pharmacy/optican/demo/service/UserService.java]: != null replace to Objects.nonNull

File: src/main/java/com/pharmacy/optican/demo/service/UserService.java [hunk 4]
```diff
@@ -3,96 +3,112 @@
...
-                    } catch (Exception e){
-                        System.out.println("Ошибка обновления в базу");
-                    }
 
+            try {
+                userRepository.save(u);
+                setAuth(u);
+            } catch (Exception e) {
+                logger.error("update error");
             }
```
Comment (EvgenKalashnikov) [inline src/main/java/com/pharmacy/optican/demo/service/UserService.java:88]: remove whitespace

File: src/main/resources/static/css/style_main.css [hunk 5]
```diff
@@ -210,6 +210,45 @@ textarea:focus {
...
+  border: 2px solid #55a831;
+  border-radius: 1rem;
+  background-color: #ffffff;
+  font-size: 1.4rem;
+  line-height: 1.2;
+  font-family: 'Open Sans', sans-serif;
+  font-weight: 700;
+  font-style: normal;
+  color: #55a831;
+  /*p {
```
Comment (EvgenKalashnikov) [inline src/main/resources/static/css/style_main.css]: remove comments

File: src/main/resources/static/css/style_main.css [hunk 6]
```diff
@@ -210,6 +210,45 @@ textarea:focus {
...
+
+  padding: 1rem 1rem 1rem 1rem;
+  font-size: 1.4rem;
+  line-height: 1.2;
+  font-family: 'Open Sans', sans-serif;
+  font-weight: 700;
+  font-style: normal;
+  color: #0089cd;
+  /*p {
+        display: inline-block;
```
Comment (EvgenKalashnikov) [inline src/main/resources/static/css/style_main.css]: the same

File: src/main/resources/static/css/style_main.css (hunk 5 above)
Comment (Main444) [reply src/main/resources/static/css/style_main.css]: DONE

File: src/main/resources/static/css/style_main.css (hunk 6 above)
Comment (Main444) [reply src/main/resources/static/css/style_main.css]: done

File: src/main/java/com/pharmacy/optican/demo/service/UserService.java (hunk 1 above)
Comment (Main444) [reply src/main/java/com/pharmacy/optican/demo/service/UserService.java:66]: done

File: src/main/java/com/pharmacy/optican/demo/service/UserService.java (hunk 3 above)
Comment (Main444) [reply src/main/java/com/pharmacy/optican/demo/service/UserService.java]: done

File: src/main/java/com/pharmacy/optican/demo/service/UserService.java (hunk 4 above)
Comment (Main444) [reply src/main/java/com/pharmacy/optican/demo/service/UserService.java:88]: done

File: src/main/java/com/pharmacy/optican/demo/service/UserService.java (hunk 2 above)
Comment (Main444) [reply src/main/java/com/pharmacy/optican/demo/service/UserService.java]: done

**Pull request merged**